text/0474-event-bridge-scheduler-l2.md

```diff
@@ -0,0 +1,481 @@
+# {RFC_TITLE}
+
+* **Original Author(s):**: @filletofish, @Jacco
+* **Tracking Issue**: #https://github.com/aws/aws-cdk-rfcs/issues/473
+* **API Bar Raiser**: @{BAR_RAISER_USER}
+
+> Write one sentence which is a brief description of the feature. It should describe:
+> * What is the user pain we are solving?
+> * How does it impact users?
+
+Library `aws-events-scheduler` contains L2 CDK constructs for creating, run, and manage scheduled tasks at scale with Amazon Event Bridge Scheduler. 
+
+
+## Working Backwards
+
+> This section should contain one or more "artifacts from the future", as if the
+> feature was already released and we are publishing its CHANGELOG, README,
+> CONTRIBUTING.md and optionally a PRESS RELEASE. This is the most important
+> section of your RFC. It's a powerful thought exercise which will challenge you
+> to truly think about this feature from a user's point of view.
+>
+> Choose *one or more* of the options below:
+>
+> * **CHANGELOG**: Write the changelog entry for this feature in conventional
+>   form (e.g. `feat(eks): cluster tags`). If this change includes a breaking
+>   change, include a `BREAKING CHANGE` clause with information on how to
+>   migrate. If migration is complicated, refer to a fictional GitHub issue and
+>   add its contents here.
+>
+> * **README**: If this is a new feature, write the README section which
+>   describes this new feature. It should describe the feature and walk users
+>   through usage examples and description of the various options and behavior.
+>
+> * **PRESS RELEASE**: If this is a major feature (~6 months of work), write the
+>   press release which announces this feature. The press release is a single
+>   page that includes 7 paragraphs: (1) summary, (2) problem, (3) solution, (4)
+>   leader quote, (5) user experience, (6) customer testimonial and (7) one
+>   sentence call to action.
+
+**CHANGELOG**:
+
+`feat(events-scheduler): Event Bridge Scheduler L2 constructs`
```

Comment:
If possible, lets break the proposed changelog into atomic pieces. Can we introduce schedule/targets/schedule-groups separately?

Reply:
Could do so, but could you please explain to me what is the value of this? We will likely commit the whole implantation in a single PR to CDK repo?

Reply:
> The value would be that its easier to review.

What exactly easier to review, the change log?

I thought this is a proposal for how the future CDK change-log would look like. See for example L2 constructs release in https://github.com/aws/aws-cdk/releases/tag/v2.65.0. In this case, isn't it easier for the reader to have a single entry for Scheduler L2 constructs rather than multiple entries for each L2 Construct Class separately? As those construct classes on their own do not provide much value, I wouldn't break them up. We could still list the new classes in the description though.

Reply:
I mean easier for me to review the code if the PRs are separated out into chunks. The smaller the PR, the less likely I am to miss things. If these constructs are not able to be separated then I understand, although I don't feel like that's necessarily the case.

For example, `ScheduleGroup` can and should be a separate PR imo. Just helps declutter the initial PR.